meteor/sofie-server/model/entities/rundown.ts

```diff
@@ -113,30 +151,46 @@ export class Rundown extends BasicRundown {
 	}
 
 	public getActiveSegment(): Segment {
-		this.assertActive('getActiveSegment')
+		this.assertActive(this.getActiveSegment.name)
 		return this.activeSegment
 	}
 
 	public getNextSegment(): Segment {
-		this.assertActive('getNextSegment')
+		this.assertActive(this.getNextSegment.name)
 		return this.nextSegment
 	}
 
 	public getActivePart(): Part {
-		this.assertActive('getActivePart')
+		this.assertActive(this.getActivePart.name)
 		return this.activePart
 	}
 
 	public getNextPart(): Part {
-		this.assertActive('getNextPart')
+		this.assertActive(this.getNextPart.name)
 		return this.nextPart
 	}
 
+	public getPreviousPart(): Part | undefined {
+		this.assertActive(this.getPreviousPart.name)
+		return this.previousPart
+	}
+
 	public takeNext(): void {
-		this.assertActive('takeNext')
+		this.assertActive(this.takeNext.name)
+		this.setPreviousPart()
 		this.takeNextPart()
 		this.takeNextSegment()
 		this.setNextFromActive()
+		this.updateInfinitePieces()
+	}
+
+	private setPreviousPart(): void {
```

Comment:
I understand what this does. The name however suggest that this is a setter method, that needs an input of type `part` to be supplied. However in reality it takes the instance property `activePart` and assigns it to the instance property `previousPart`.

Perhabs renaming it to `shiftActiveToPreviousPart()`, or something similar could be of help.

Reply:
It is a setter method. It just doesn't get the value to set externally.

Reply:
Fair enough.